README.md

```diff
@@ -1,10 +1,109 @@
-azure-chef-extension
-====================
+#azure-chef-extension
 
 Azure resource extension to enable Chef on Azure virtual machine instances.
 
-Azure Chef Extension Version Scheme
-===================================
+##Features:
+
+1. It can be installed through Azure RESTFUL API for extension
+2. The execution output of the scripts is logged in the log directory specified in HandlerEnvironment.json
+3. Status of the extension is reported back to Azure so that user can see the status on Azure Portal
+
+##Platforms and version its supported:
+
+| Platform | Version    |
+|----------|------------|
+| Ubuntu   | 12.04, 14+  |
+| Windows  | 2008r2, 2012, 2012r2 |
+
+
+##Azure Chef Extension usage:
+#####Options those can be set in publicconfig.config
+```javascript
+{
+  "client_rb": "< your client.rb configuration >".
+  "runlist":"< your run list >",
+  "autoUpdateClient":"< true|false >",
+  "deleteChefConfig":"< true|false >",
+  "bootstrap_options": {
+    "chef_node_name":"< your node name >",
+    "chef_server_url":"< your chef server url >",
+    "validation_client_name":"< your chef organization validation client name  >"
+  }
+}
+```
+`client_rb`: Set this option to specify the configuration details for the chef-client. Refer to [client.rb] (https://docs.chef.io/config_rb_client.html)
+
+`run_list`: A run-list defines all of the information necessary for Chef to configure a node into the desired state.
+It is an ordered list of roles and/or recipes that are run in the exact order defined in the run-list.
+
+`autoUpdateClient` : Set this option to true to auto udpate chef extension version by default its set to false.  Extension's Hotfix versions are auto-updated on the VM when the VM is restarted. for e.g.: A VM that has 1205.12.2.0, gets auto updated to 1205.12.2.1 when 1205.12.2.1 is published.
+
+`deleteChefCofig`: Set deleteChefConfig option to true in publicconfig if you want to delete chef configuration files while update or uninstall. By default it is set to false.
+
+`bootstrap_options`: Set bootstrap options while creating a VM through Azure using powershell cmdlets. Bootstrap options used by Chef-Client during node converge. It overrides the configuration set in client_rb option. for e.g. node_name option i.e. if you set node_name as "foo" in the client_rb and in bootstrap_option you set chef_node_name as "bar" it will take "bar" as node name instead of "foo".
```

Comment:
Could you please replace above `Set bootstrap options while creating a VM through Azure using powershell cmdlets` line by `Set bootstrap options while adding chef extension to Azure VM` . The use of 'powershell' in above options description is unnecessary there as user can try different CLI tools (like azure-xplat-cli, PowerShell, knife-azure etc) to create VM with extension .